codec-http/src/main/java/io/netty5/handler/codec/http/headers/DefaultHttpHeaders.java

```diff
@@ -81,9 +82,52 @@ public DefaultHttpHeaders(final int arraySizeHint, final boolean validateNames,
         this.validateValues = validateValues;
     }
 
+    @Override
+    public boolean contains(CharSequence key, CharSequence value) {
+        return contains(key, value, DefaultHttpHeaders::containsCommaSeparatedTrimmedCaseSensitive);
+    }
+
     @Override
     public boolean containsIgnoreCase(final CharSequence name, final CharSequence value) {
-        return contains(name, value, AsciiString::contentEqualsIgnoreCase);
+        return contains(name, value, DefaultHttpHeaders::containsCommaSeparatedTrimmedCaseInsensitive);
+    }
+
+    private static boolean containsCommaSeparatedTrimmedCaseSensitive(CharSequence expected, CharSequence rawNext) {
+        int begin = 0;
+        int end;
+        if ((end = AsciiString.indexOf(rawNext, ',', begin)) == -1) {
+            return contentEquals(trim(rawNext), expected);
+        }
+        do {
+            if (contentEquals(trim(rawNext.subSequence(begin, end)), expected)) {
+                return true;
+            }
+            begin = end + 1;
+        } while ((end = AsciiString.indexOf(rawNext, ',', begin)) != -1);
+
+        if (begin < rawNext.length()) {
+            return contentEquals(trim(rawNext.subSequence(begin, rawNext.length())), expected);
+        }
+        return false;
+    }
+
+    private static boolean containsCommaSeparatedTrimmedCaseInsensitive(CharSequence expected, CharSequence rawNext) {
+        int begin = 0;
+        int end;
+        if ((end = AsciiString.indexOf(rawNext, ',', begin)) == -1) {
+            return contentEqualsIgnoreCase(trim(rawNext), expected);
+        }
+        do {
+            if (contentEqualsIgnoreCase(trim(rawNext.subSequence(begin, end)), expected)) {
+                return true;
+            }
+            begin = end + 1;
+        } while ((end = AsciiString.indexOf(rawNext, ',', begin)) != -1);
+
+        if (begin < rawNext.length()) {
+            return contentEqualsIgnoreCase(trim(rawNext.subSequence(begin, rawNext.length())), expected);
+        }
+        return false;
```

Comment:
nit: you could share some code here by having the method take another boolean which influence if either `contentEquals` or `contentEqualsIgnoreCase` is used.